src/linker/Linker/CompilerGeneratedState.cs

```diff
@@ -129,9 +142,24 @@ void ProcessMethod (MethodDefinition method)
 			// code or its referenced nested functions. There is no reliable way to determine from
 			// IL which user code an unused nested function belongs to.
 			foreach (var userDefinedMethod in callingMethods) {
-				foreach (var nestedFunction in callGraph.GetReachableMethods (userDefinedMethod)) {
-					Debug.Assert (CompilerGeneratedNames.IsLambdaOrLocalFunction (nestedFunction.Name));
-					_compilerGeneratedMethodToUserCodeMethod.Add (nestedFunction, userDefinedMethod);
+				foreach (var compilerGeneratedMember in callGraph.GetReachableMembers (userDefinedMethod)) {
+					switch (compilerGeneratedMember) {
+					case MethodDefinition nestedFunction:
+						Debug.Assert (CompilerGeneratedNames.IsLambdaOrLocalFunction (nestedFunction.Name));
+						// Nested functions get suppressions from the user method only.
+						if (!_compilerGeneratedMethodToUserCodeMethod.TryAdd (nestedFunction, userDefinedMethod)) {
+							var alreadyAssociatedMethod = _compilerGeneratedMethodToUserCodeMethod[nestedFunction];
+							_context.LogWarning (new MessageOrigin (userDefinedMethod), DiagnosticId.MethodsAreAssociatedWithUserMethod, userDefinedMethod.GetDisplayName (), alreadyAssociatedMethod.GetDisplayName (), nestedFunction.GetDisplayName ());
+						}
+						break;
+					case TypeDefinition stateMachineType:
+						// State machine types get suppressions from the state machine method,
+						// already tracked above.
```

Comment:
```suggestion
						// Types in the call graph are always state machine types
						// For those all their methods are also part of the same call graph (the type is considered to be "transparent" and we track references/calls through it)
						// So we can rely on the graph to report all of the state machine type's methods to be reported
						// as part of the "reachable" members anywhere the state machine type itself is used.
```

It took me a long time to realize what this does and what the comment means. Something more descriptive in the comment would help a lot I think.

Reply:
I adjusted the comment slightly, hopefully it's more clear now. Thanks!